kernel/src/table.rs

```diff
@@ -106,12 +112,12 @@ fn resolve_uri_type(table_uri: impl AsRef<str>) -> DeltaResult<UriType> {
         } else if scheme.len() == 1 {
             // NOTE this check is required to support absolute windows paths which may properly
             // parse as url we assume here that a single character scheme is a windows drive letter
-            Ok(UriType::LocalPath(PathBuf::from(table_uri)))
+            Ok(UriType::LocalPath(PathBuf::from(table_uri.as_ref())))
```

Comment:
side question:" Why does this `LocalPath` use `table_uri` but the other (L108) uses `url.to_file_path`? Is that because we don't have a `file://` scheme here?

Reply:
Yeah. True to form, the URL crate isn't really helpful. From the [`to_file_path`](https://docs.rs/url/latest/url/struct.Url.html#method.to_file_path) docs:
> Assuming the URL is in the file scheme or similar, convert its path to an absolute std::path::Path.
> Note: This does not actually check the URL’s scheme, and may give nonsensical results for other schemes. It is the user’s responsibility to check the URL’s scheme before calling this.

So this seems safer.